#60 Release Notes v1.1.0 (#61)
Inbibo/adonisfx_docs: docs/release_notes.md

Reviewer: I want to ask you about having a look at "Known Limitations" because a lot of them might be resolved!

((I'm not sure if it should be changed in this issue.))

Author: I think we can modify the Known Limitations in this issue since we are modifying `release_notes.md`. I've removed:

- The AdnLocatorRotation does not draw the angle shape if the three inputs (start, mid and end positions) are aligned, i.e. when the angle is 180 degrees. *AdonisFX-1000*
- The debugger system draws debug data independently to the visibility of the mesh. *AdonisFX-802*

Regarding the limitation about the ‘Maximum sliding distance’ it is true that we have added the warning dialog, however the sliding constraints will still be time and memory consuming for high values, so I don't know if we should remove it or modify it.

Please, let me know if I'm wrong or if I should remove any other known limitation.

Author: Totally right, removed those limitations.

# Release Notes

## Version 1.1.0

> ### Bug Fixes

Reviewer 2: Move bug fixes after Features

- Filtered characters to get valid identifiers in Maya for Linux. *AdonisFX-1033*
- Fixed element selection check at node creation. *AdonisFX-1031*
- Fixed error messages not displaying the proper status of the AdnSimshape collider connections. *AdonisFX-1034*

> - Prevent AAD file export if there is nothing to export or the only thing to export failed. *AdonisFX-1032*

Reviewer 2: ```suggestion
- Prevent Exporter to generate the AAD file export if there is no valid data to export.
```

- Fixed manipulation of referenced nodes while using the AdonisFX paint tool. *AdonisFX-1147*
- Fixed a bug that did not allow to add attachments on a muscle deformer created on referenced geometry. *AdonisFX-506*
- Fixed import of the constraint weight arrays on sparse arrays. *AdonisFX-1155*
- Fixed a bug that caused geometry sliding constraints to not find valid triangles to slide on initialization. *AdonisFX-1165*

> - Fixed paint flood operation not working correctly for multi influence constraints. *AdonisFX-1146*

Reviewer 2: ```suggestion
- Fixed paint flood operation not normalizing the weights properly. *AdonisFX-1146*
```

Author: Done.

> ### Features

Reviewer 2: In the section of "New Features" I would avoid to use the AdonisFX issue number. In many cases, the features will be a compound of issues

Author: Removed issue number from notes related to new features.

> ### Features

Reviewer 2: Create a section "Core" where you add:
- attach to geo
- slide on geo
- mass support in muscle solvers

OF course, with a proper description

Author: Added "Core" section before the "Maya" section.

> ### Features

Reviewer 2: ```suggestion
### Maya
```

Author: Renamed "Features" section as "Maya".

> - Extended AdonisFX menu with *Documentation*, *Tutorials* and *About* sections. *AdonisFX-1095*

Reviewer 2: At the end

Author: Done.

- Added the ability to prevent the debugging of features if the geometry is not visible, hidden or isolated. *AdonisFX-802*

> - Added a warning dialog in AdnSkin and AdnSimshape when the *Max Sliding Distance* is considered too high for a mesh, which could freeze Maya. *AdonisFX-997*, *AdonisFX-466*

Reviewer 2: ```suggestion
- Added a sanity check to *Max Sliding Distance* attribute to warn the user if the input value is too high for the given target mesh.
```

Author: Done.

> - Added the ability to debug the *Sliding Surface* in AdnSkin and AdnSimshape. *AdonisFX-997*, *AdonisFX-466*

Reviewer 2: Rewrite this to inform that the debugger has been extended to support: sliding surface, attachment to geo, slide on geo

Author: Done.

> - Added attachment to geometry constraints to AdnMuscle and AdnRibbonMuscle, allowing the muscle to be attached directly to a geometry. *AdonisFX-1081*

Reviewer 2: Move it at the top, after AdnSkinMerge

Author: Done.

> - Added attachment to geometry constraints to AdnMuscle and AdnRibbonMuscle, allowing the muscle to be attached directly to a geometry. *AdonisFX-1081*

Reviewer 2: ```suggestion
- Added support for attachment to geometry in AdnMuscle and AdnRibbonMuscle solvers.
```

Author: Done.

> - Added a new AdonisFX deformer in charge of merging the results given by the skin simulation mesh and the animation mesh. *AdonisFX-1103*

Reviewer 2: Move this one at the top

```suggestion
- Added AdnSkinMerge deformer to blend the results from animated and simualted input geometries.
```

Author: Done.

> - Added slide on geometry constraints to AdnMuscle and AdnRibbonMuscle, allowing the muscle to slide directly on a geometry. *AdonisFX-1140*

Reviewer 2: Move it at the top, after attach to geo

> - Added slide on geometry constraints to AdnMuscle and AdnRibbonMuscle, allowing the muscle to slide directly on a geometry. *AdonisFX-1140*

Reviewer 2: ```suggestion
- Added support to slide on geometry in AdnMuscle and AdnRibbonMuscle solvers.
- Added support to multiple geometry targets in AdnMuscle and AdnRibbonMuscle solvers for attachments and sliding.
```

> - Added mass support for AdnMuscle and AndRibbonMuscle deformers, allowing better control to achieve the desired results. *AdonisFX-1079*

Reviewer 2: Move it after slide on geo

```suggestion
- Added mass attributes in AdnMuscle and AndRibbonMuscle deformers to provide more control over the dynamics of the simulated object.
```

Author: Done.

> - Added *Point Mass Mode* to AdonisFX deformers, which allows to specify either a mass per point or a density value from which point masses are computed. *AdonisFX-1079*

Reviewer 2: ```suggestion
- Added *Point Mass Mode* to AdnSimshape, AdnSkin, AdnMuscle and AdnRibbonMuscle which allows to specify either a mass per point or a density value from which point masses will be computed.
```

> - Added importer/exporter support for AdnSkinMerge deformer and new constraints: attachment to geometry and slide on geometry. *AdonisFX-1142*

Reviewer 2: ```suggestion
- Extended Importer and Exporter tools to support AdnSkinMerge deformer and new constraints (attachment to geometry and slide on geometry).
```

# Release Notes

## Version 1.1.0

### Core

> - Attachment to geometry constraints to allow muscles to be attached to any given geometry.

Author: Done.

> - Filtered characters to get valid names for nodes and attributes since some special characters are not allowed in Maya for Linux. *AdonisFX-1033*

Reviewer: ```suggestion
- Filtered characters to get valid identifiers in Maya for Linux. *AdonisFX-1033*
```

Author: More concise, done!

> - Slide on geometry constraints to allow muscles to slide on any given geometry.

Author: Done.

> - Fixed Python runtime error due to incorrect element selection on node creation. *AdonisFX-1031*

Reviewer: I would not say "runtime errors", for me seems that we are giving a bad a image of the product.

```suggestion
- Fixed element selection check at node creation. *AdonisFX-1031*
```

Author: I agree, done.

> - Mass support in muscle solvers to provide more control over the dynamics of the simulation.

Author: Done.

> - Fixed error messages not displaying the proper status of the AdnSimshape connections. *AdonisFX-1034*

Reviewer: ```suggestion
- Fixed error messages not displaying the proper status of the AdnSimshape collider connections. *AdonisFX-1034*
```

Author: Done.

> - Fixed exporter creating the AAD file when there is nothing to export or the only thing to export failed. *AdonisFX-1032*

Reviewer: ```suggestion
- Prevent AAD export if there is nothing to export or the only thing to export failed. *AdonisFX-1032*
```

Author: I finally modified it to :
```suggestion
- Prevent AAD file export if there is nothing to export or the only thing to export failed. *AdonisFX-1032*
```

- Fixed manipulation of referenced nodes while using the AdonisFX paint tool. *AdonisFX-1147*

> - Fixed a bug that did not allow to add attachments on a AdnMuscle/AdnRibbonMuscle deformer created on referenced geometry. *AdonisFX-506*

Reviewer: ```suggestion
- Fixed a bug that did not allow to add attachments on a muscle deformer created on referenced geometry. *AdonisFX-506*
```

Author: Agreed, done.

- Added AdnSkinMerge deformer to blend the results from animated and simualted input geometries.
- Added support for attachment to geometry in AdnMuscle and AdnRibbonMuscle solvers.
- Added support to slide on geometry in AdnMuscle and AdnRibbonMuscle solvers.
- Added mass attributes in AdnMuscle and AndRibbonMuscle deformers to provide more control over the dynamics of the simulated object.

> - Added the ability to prevent the debugging of features if the geometry is not visible, hidden or isolated.

Reviewer 2: ```suggestion
- Debugger system improved to be dependent to the visibility of the simulated geometry.
```

Author: Done.

> - Fixed a bug in the handling of the constraint weight arrays. *AdonisFX-1155*

Reviewer: ```suggestion
- Fixed import of the constraint weight arrays on sparse arrays. *AdonisFX-1155*
```

Author: More precise, done.

> - Fixed a bug that caused geometry sliding constraints to not find valid triangles to slide on for some vertices in very specific scenarios. *AdonisFX-1165*

Reviewer: I would not add the part of "very specific scenarios". Instead I would point where the change is affecting.

```suggestion
- Fixed a bug that caused geometry sliding constraints to not find valid triangles to slide on initialization. *AdonisFX-1165*
```

Author: Done.

> - Fixed flood operation not working correctly for multi influence constraints. *AdonisFX-1146*

Reviewer: ```suggestion
- Fixed paint flood operation not working correctly for multi influence constraints. *AdonisFX-1146*
```

Author: Added.

> - Optimzed the initialization of the debugger improving the performance on scenes with many AdonisFX deformers. *AdonisFX-1167*

Reviewer: There are some things to say there.
1. This improvement was removed as the method where it was had a lot of complexity that got reduced drastically by the new utility. So, I would remove this line.
2. We added a new utility to AdonisFX menu to help with the debugging graph connections. #1171. That is not present. Here I would mention the utility and improvement.
3. "Optimized"

Author: 1. Removed this improvement.
2. Added #1171 (I think it was not merged when I did the first list) as improvement. However the issue is tagged as feature, should we move it to feature?

Reviewer: We should move it to features, yes!

Author: Moved to features.

> - Added *Point Mass Mode* to AdnSimshape, AdnSkin, AdnMuscle and AdnRibbonMuscle which allows to specify either a mass per point or a density value from which point masses will be computed.

Reviewer 2: Please, move this below "Added mass attributes..."

Author: Done.

### Features

- Extended AdonisFX menu with *Documentation*, *Tutorials* and *About* sections. *AdonisFX-1095*
- Added the ability to prevent the debugging of features if the geometry is not visible, hidden or isolated. *AdonisFX-802*

> - Added a warning dialog in AdnSkin when the *Max Sliding Distance* is considered too high for a mesh, which could freeze Maya. *AdonisFX-997*
> - Added the ability to debug the *Sliding Surface* in AdnSkin. *AdonisFX-997*
> - Added a warning dialog in AdnSimshape when the *Max Sliding Distance* is considered too high for a mesh, which could freeze Maya. *AdonisFX-466*
> - Added the ability to debug the *Sliding Surface* in AdnSimshape. *AdonisFX-466*

Reviewer: ```suggestion
- Added a warning dialog in AdnSkin and AdnSimshape when the *Max Sliding Distance* is considered too high for a mesh, which could freeze Maya. *AdonisFX-997*, *AdonisFX-466*
- Added the ability to debug the *Sliding Surface* in AdnSkin and AdnSimshape. *AdonisFX-997*, *AdonisFX-466*
```

Author: I like it, added.

- Fixed error messages not displaying the proper status of the AdnSimshape collider connections. *AdonisFX-1034*

> - Prevent Exporter to generate the AAD file export if there is no valid data to export. *AdonisFX-1032*

Reviewer 2: ```suggestion
- Prevent Exporter to generate the AAD file if there is no valid data to export. *AdonisFX-1032*
```

Author: Done.

> - Added a new AdonisFX deformer in charge of merging the results given by skin simulation mesh and animation mesh into a final mesh. *AdonisFX-1103*

Reviewer: ```suggestion
- Added a new AdonisFX deformer in charge of merging the results given by skin simulation mesh and the animation mesh. *AdonisFX-1103*
```

> - Added slide on geometry constraints to AdnMuscle and AdnRibbonMuscle, allowing the muscle to slide directly on the geometry of the bone. *AdonisFX-1140*

Reviewer: ```suggestion
- Added slide on geometry constraints to AdnMuscle and AdnRibbonMuscle, allowing the muscle to slide directly on a geometry. *AdonisFX-1140*
```

Author: Modifed also for attach to geo constraints.

- Added mass support for AdnMuscle and AndRibbonMuscle deformers, allowing better control to achieve the desired results. *AdonisFX-1079*
- Added *Point Mass Mode* to AdonisFX deformers, which allows to specify either a mass per point or a density value from which point masses are computed. *AdonisFX-1079*

> - Added importer/exporter support for attachment to geometry and slide on geometry constraints. *AdonisFX-1142*
> - Added importer/exporter support for the AdnSkinMerge deformer. *AdonisFX-1142*

Reviewer: ```suggestion
- Added importer/exporter support for attachment to geometry and slide on geometry constraints; AdnSkinMerge deformer. *AdonisFX-1142*
```

Author: I've combined those lines into this one:
```suggestion
- Added importer/exporter support for AdnSkinMerge deformer and new constraints: attachment to geometry and slide on geometry. *AdonisFX-1142*
```

- The *Max Sliding Distance* parameter in AdnSkin and AdnSimshape is represented in scene units. The higher this value is, the more units in the space the sliding constraint allows a vertex to slide on. If the polygons of the target geometry to slide on are very little compared to that value, then the sliding constraint will be time and memory consuming. *AdonisFX-997*
- The activations debugger in AdnSimshape is limited to the Parallel and Serial evaluation modes. *AdonisFX-1026*

> - The debugger system draws debug data independently to the visibility of the mesh. *AdonisFX-802*

Author: I will add them again, I didn't thought of that.
